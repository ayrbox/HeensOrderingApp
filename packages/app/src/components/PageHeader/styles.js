const styles = theme => ({
  header: {
    marginBottom: `${theme.spacing.unit * 4}px`,
  },
});

export default styles;
