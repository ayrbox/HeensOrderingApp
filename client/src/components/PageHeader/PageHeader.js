import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import AddIcon from '@material-ui/icons/Add';
import Typography from '@material-ui/core/Typography';
import Fab from '@material-ui/core/Fab';

import styles from './styles';

const PageHeader = ({
  title,
  subTitle,
  classes,
  onAddClicked,
}) => (
  <Grid
    container
    direction="row"
    justify="space-between"
    alignItems="center"
    className={classes.header}
  >
    <Grid item>
      <Typography variant="h1">{title}</Typography>
      {subTitle && <Typography variant="subtitle1">{subTitle}</Typography>}
    </Grid>
    {onAddClicked && (
      <Grid item>
        <Fab
          color="primary"
          aria-label="Add"
          size="small"
          onClick={onAddClicked}
        >
          <AddIcon />
        </Fab>
      </Grid>
    )}
  </Grid>
);

PageHeader.defaultProps = {
  subTitle: null,
  onAddClicked: undefined
};

PageHeader.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  classes: PropTypes.shape().isRequired,
  onAddClicked: PropTypes.func,
};

export default withStyles(styles)(PageHeader);
