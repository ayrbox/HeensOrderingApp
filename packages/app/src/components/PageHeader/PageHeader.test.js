import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import PageHeader from './PageHeader';
import '@testing-library/jest-dom/extend-expect';

describe('<PageHeader />', () => {
  const mockAdd = jest.fn();

  const makePageHeader = (
    title = 'Page Title',
    subtitle = 'Sub Title',
    onAdd = mockAdd,
  ) => render(
    <PageHeader
      title={title}
      subTitle={subtitle}
      onAddClicked={onAdd}
    />,
  );

  afterEach(() => {
    cleanup();
    mockAdd.mockClear();
  });

  it('renders correctly', () => {
    const { container } = makePageHeader();
    expect(container).toMatchSnapshot();
  });

  it('renders title and sub title', () => {
    const { getByTestId } = makePageHeader();
    const h1 = getByTestId('mainTitle');
    expect(h1).toHaveTextContent('Page Title');

    expect(getByTestId('sub-title')).toHaveTextContent('Sub Title');
  });

  it('fires click methods', () => {
    const { getByTestId } = makePageHeader();
    fireEvent.click(getByTestId('btn-add'));

    expect(mockAdd).toHaveBeenCalled();
  });

  describe('when subtitle and add func not supplied', () => {
    it('should not content subtitle and add button', () => {
      const { queryByTestId } = render(
        <PageHeader title="here is a title" />,
      );

      expect(queryByTestId('sub-title')).toBeNull();
      expect(queryByTestId('btn-add')).toBeNull();
    });
  });
});
