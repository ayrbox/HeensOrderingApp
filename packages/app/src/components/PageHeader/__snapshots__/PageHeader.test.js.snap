// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-container-2 MuiGrid-align-items-xs-center-10 MuiGrid-justify-xs-space-between-21 PageHeader-header-1"
  >
    <div
      class="MuiGrid-item-3"
    >
      <h1
        class="MuiTypography-root-99 MuiTypography-h1-111"
        data-testid="mainTitle"
      >
        Page Title
      </h1>
      <h6
        class="MuiTypography-root-99 MuiTypography-subtitle1-117"
        data-testid="sub-title"
      >
        Sub Title
      </h6>
    </div>
    <div
      class="MuiGrid-item-3"
    >
      <button
        aria-label="Add"
        class="MuiButtonBase-root-145 MuiFab-root-135 MuiFab-primary-137 MuiFab-sizeSmall-143"
        data-testid="btn-add"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiFab-label-136"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-148"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-157"
        />
      </button>
    </div>
  </div>
</div>
`;
